chore: periodically resolve trusted nodes dns

shane-moore opened the pull request:
## Changes
This PR attempts to close https://github.com/paradigmxyz/reth/issues/15623. I'm super open to any feedback or suggestions to improve the implementation!

mattsse reviewed (changes requested):
great, this looks feature complete and correct, only have cosmetic nits

mattsse commented on crates/net/network/src/peers.rs:
```diff
@@ -35,6 +36,57 @@ use tokio::{
+    /// Poll the resolver.
+    /// When the interval ticks, new resolution futures for each trusted peer are spawned.
+    /// If a future completes successfully, it returns the resolved (`PeerId`, `NodeRecord`).
+    pub fn poll(&mut self, cx: &mut Context<'_>) -> Poll<Option<(PeerId, NodeRecord)>> {
```
we can shortcurcuit here if the trusted peers are empty

shane-moore commented on crates/net/network/src/peers.rs:
```diff
@@ -35,6 +36,57 @@ use tokio::{
+    pub interval: Interval,
+    /// Futures for currently in‑flight resolution tasks.
+    pub pending: FuturesUnordered<BoxFuture<'static, (PeerId, Result<NodeRecord, io::Error>)>>,
+}
```
💯

shane-moore commented on crates/net/network/src/peers.rs:
```diff
@@ -35,6 +36,57 @@ use tokio::{
+            }
+        }
+
+        match self.pending.poll_next_unpin(cx) {
```
luv this!

shane-moore commented on crates/net/network/src/peers.rs:
```diff
@@ -990,6 +1049,20 @@ impl PeersManager {
+                        peer.addr = new_addr;
+                        trace!(target: "net::peers", ?peer_id, "Updated trusted peer address via DNS resolution");
+                    }
+                }
```
🫡

shane-moore commented:
@mattsse, addressed PR review comments in latest commit https://github.com/paradigmxyz/reth/pull/15678/commits/a444a349d15bac9c4d31d405f0c0f22911ee0c86

lmk if any nits on the commit as well. appreciate the feedback so far 🎉

mattsse reviewed (approved):
ty!
